packages/component-base/src/polylit-mixin.js

```diff
@@ -103,6 +103,22 @@ const PolylitMixinImplementation = (superclass) => {
         });
       }
 
+      if (options.sync) {
```

Comment:
nit: Hypothetically it should work even if we made this the default (removed the check here)? I'm getting some errors however `Cannot set properties of undefined (setting 'isConnected')`. Probably this is a no-issue since it's an opt-in feature.

Reply:
Yes, this error actually happened when I tried this in a real component e.g. `vaadin-infinite-scroller`.
I will continue investigating before merging this PR, could be a timing issue.

Reply:
When all the properties are synchronous, will it force the component to render synchronously during initialization?